subroutines/habits/models/instances.py

```diff
@@ -0,0 +1,56 @@
+from datetime import timedelta
+
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+
+class InstanceManager(models.Manager):
+    """Instance Manager."""
+
+    def get_stats_per_habit(self, habit, date):
+        """Returns the total of instances both done and don't per habit."""
+        query = self.filter(habit=habit, date_to_do__lte=date)
+        return {
+            "total": query.count(),
+            "total_done": query.filter(is_done=True).count(),
+        }
+
+    def get_stats_per_user(self, user):
+        """
+        Returns both the total of habit instances a user should have
+        been done yesterday and total of instances done.
+        """
+        yesterday = timezone.datetime.today() - timedelta(days=1)
+        query = self.filter(user=user, date_to_do=yesterday)
+        return {
+            "total": query.count(),
+            "total_done": query.filter(is_done=True).count(),
+        }
+
+    def get_for_today(self, user):
+        """
+        Returns a list of habit instances that a user must to do today
+        excluding instances of habits paused.
+        """
+        today = timezone.datetime.today()
+        return self.filter(user=user, date_to_do=today, habit__is_paused=False)
+
+
+class Instance(models.Model):
+
+    date_to_do = models.DateField(
+        _("date"),
+        default=timezone.now,
+        help_text=_("Date at which an habit must be done"),
+    )
+    is_done = models.BooleanField(default=False)
+
+    habit = models.ForeignKey("habits.Habit", on_delete=models.DO_NOTHING)
+    user = models.ForeignKey("users.User", on_delete=models.DO_NOTHING)
```

Comment:
All do nothing might cause integrity error. Setting to null might be an option you'd like to consider.